examples/options/src/options/throughput_options.cpp

```diff
@@ -0,0 +1,74 @@
+/** TRACCC library, part of the ACTS project (R&D line)
+ *
+ * (c) 2022 CERN for the benefit of the ACTS project
+ *
+ * Mozilla Public License Version 2.0
+ */
+
+// Library include(s).
+#include "traccc/options/throughput_options.hpp"
+
+// System include(s).
+#include <iostream>
+#include <stdexcept>
+
+namespace traccc {
+
+namespace po = boost::program_options;
+
+throughput_options::throughput_options(po::options_description& desc) {
+
+    desc.add_options()("input_data_format",
+                       po::value<std::string>()->default_value("csv"),
+                       "Format of the input file(s)");
+    desc.add_options()("input_directory", po::value<std::string>()->required(),
+                       "Directory holding the input files");
+    desc.add_options()("detector_file", po::value<std::string>()->required(),
+                       "Detector geometry description file");
+    desc.add_options()("digitization_config_file",
+                       po::value<std::string>()->required(),
+                       "Digitization configuration file");
+    desc.add_options()("loaded_events",
+                       po::value<std::size_t>()->default_value(10),
+                       "Number of input events to load");
+    desc.add_options()("processed_events",
+                       po::value<std::size_t>()->default_value(100),
+                       "Number of events to process");
+}
+
+void throughput_options::read(const po::variables_map& vm) {
+
+    // Set the input data format.
+    if (vm.count("input_data_format")) {
+        const std::string input_format_string =
+            vm["input_data_format"].as<std::string>();
+        if (input_format_string == "csv") {
+            input_data_format = data_format::csv;
+        } else if (input_format_string == "binary") {
+            input_data_format = data_format::binary;
+        } else {
+            throw std::invalid_argument("Invalid input data format specified");
+        }
+    }
+
+    // Set the rest of the options.
+    input_directory = vm["input_directory"].as<std::string>();
+    detector_file = vm["detector_file"].as<std::string>();
+    digitization_config_file = vm["digitization_config_file"].as<std::string>();
+    loaded_events = vm["loaded_events"].as<std::size_t>();
+    processed_events = vm["processed_events"].as<std::size_t>();
```

Comment:
I was instead pondering about adding some checks here. :thinking: But in the end decided against that... (Misconfigurations in these values should be fairly obvious as soon as the job starts running.)